src/frequenz/sdk/timeseries/_quantities.py

```diff
@@ -606,7 +606,15 @@ def as_megawatts(self) -> float:
         """
         return self._base_value / 1e6
 
-    @overload  # type: ignore
+    # We need the ignore here because otherwise mypy will give this error:
+    # > Overloaded operator methods can't have wider argument types in overrides
+    # The problem seems to be when the other type implements an **incompatible**
+    # __rmul__ method, which is not the case here, so we should be safe.
+    # Please see this example:
+    # https://github.com/python/mypy/blob/c26f1297d4f19d2d1124a30efc97caebb8c28616/test-data/unit/check-overloading.test#L4738C1-L4769C55
+    # And a discussion in a mypy issue here:
+    # https://github.com/python/mypy/issues/4985#issuecomment-389692396
+    @overload  # type: ignore[override]
```

Comment:
Just to add some more context, when Marenz was adding these, we had no idea what the issue was, because without these, `mypy` kept crashing.  Looks like they've fixed that bug and now we are able to understand the issue.

Reply:
Yeah, I didn't get a crash (thankfully!) but I still had to do my research, because the error message is not very clear why is there a problem with doing that.